src/charm.py

```diff
@@ -367,18 +371,26 @@ def _make_env_config(self, domain="svc.cluster.local") -> dict:
         """
         config = self.model.config
         relation = self.model.get_relation("ingress-proxy")
-        if relation:
-            site = relation.data[relation.app]["service-hostname"]
-            svc_name = relation.data[relation.app]["service-name"]
-            svc_port = relation.data[relation.app]["service-port"]
-            backend_site_name = relation.data[relation.app]["service-hostname"]
+        if relation is not None and relation.data[relation.app] and relation.units:
+            nginx_config_list = ["service-hostname", "service-name", "service-port"]
```

Comment:
I think this is the same as REQUIRED_INGRESS_RELATION_FIELDS from the library? Can we use that instead?